formats_err/imports_enum_leaking.ksy

```diff
@@ -0,0 +1,15 @@
+# ..\tests\formats_err/imports/enum_two.ksy: /seq/0/enum:
```

Comment:
Should this be
```suggestion
# imports/enum_two.ksy: /seq/0/enum:
```
?
Or
```suggestion
# ../tests/formats_err/imports/enum_two.ksy: /seq/0/enum:
```
?

Reply:
The first option, because that's what you get on Unix-like systems. The fact that it's not the case on Windows can be considered a minor bug (although it's just a detail in the error message, so it's a bit annoying, but relatively harmless).

I also develop mostly on Windows (I run some things in WSL 2, but not all), so this sort of affects my local setup as well, but I'm relatively fine with ignoring a few local path-related test failures. If you want to resolve this properly, check out this commit of mine first, which will give you an explanation how these paths in `formats_err` error messages work: https://github.com/kaitai-io/kaitai_struct_compiler/commit/5bc871a79afc693ca08bf6109416ce019a919cb1

The thing is that the Unix path would only contain forward slashes (like `../tests/formats_err/imports/enum_two.ksy: /seq/0/enum:`), in which case `../tests/formats_err/` gets removed and only `imports/enum_two.ksy: /seq/0/enum:` remains. After taking a quick look, this apparently doesn't happen on Windows because of the way how `yamlDir` is computed here in KSC ([`JavaKSYParser.scala:21-22`](https://github.com/kaitai-io/kaitai_struct_compiler/blob/2191070c7a8a7e0c19b01ca79f5616437cc7bc6f/jvm/src/main/scala/io/kaitai/struct/formats/JavaKSYParser.scala#L21-L22)):

```scala
      val yamlDir = Option(new File(yamlFilename).getParent).getOrElse(".")
      val specs = new JavaClassSpecs(yamlDir, config.importPaths, firstSpec)
```

The Java `File` class notoriously _normalizes_ all paths to backslash-separated `\` on Windows (more on that here: https://github.com/kaitai-io/kaitai_struct/issues/507#issuecomment-1679686055), so it isn't very surprising to me that the value of `yamlDir` is `..\tests\formats_err` even though `yamlFilename` was `../tests/formats_err/imports_enum_leaking.ksy`.

The fix would be to use some other way to get the directory name of `yamlFilename` than `new File(yamlFilename).getParent` (as far as I know, the path normalization of Java `File` class can't be disabled, but there may be other APIs that don't do it). I'll just note that we shouldn't use any custom string manipulation for this (because we'd probably mess it up), but really try to find some existing reliable API in the standard library first.

Reply:
Ok, I removed the prefix and fix stripping on Windows in https://github.com/kaitai-io/kaitai_struct_compiler/pull/304